management/operatingsystemimage/client.go

```diff
@@ -0,0 +1,57 @@
+package operatingsystemimage
+
+import (
+	"encoding/xml"
+	"errors"
+	"fmt"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+)
+
+const (
+	azureImageListURL    = "services/images"
+	errInvalidImage      = "Can not find image %s in specified subscription, please specify another image name."
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+//NewClient is used to instantiate a new OsImageClient from an Azure client
+func NewClient(client management.Client) OsImageClient {
+	return OsImageClient{client: client}
+}
+
+func (self OsImageClient) GetImageList() ([]OSImage, error) {
+	imageList := imageList{}
+
+	response, err := self.client.SendAzureGetRequest(azureImageListURL)
+	if err != nil {
+		return imageList.OSImages, err
+	}
+
+	err = xml.Unmarshal(response, &imageList)
+	if err != nil {
+		return imageList.OSImages, err
+	}
+
+	return imageList.OSImages, err
+}
+
+func (self OsImageClient) ResolveImageName(imageName string) error {
```

Comment:
this method does not match a unique REST call, rather implements a special client-side logic. we don't want these in this library. e.g. I might wanna implement a case-insensitive comparison with this but then this method becomes useless. unless very necessary, let's drop helpers.